### PR TITLE
Support for pcf user-agent when querying for updates.

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

Minio PCF startup script should set MINIO_PCF=1 in the env var which will be used for constructing user-agent.

## How Has This Been Tested?

No straight forward way of testing this fix.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/update-main.go [hunk 1]
```diff
@@ -169,6 +169,12 @@ func IsKubernetes() bool {
 	return os.Getenv("KUBERNETES_SERVICE_HOST") != ""
 }
 
+// IsPCF returns true if minio is running on Pivotal Cloud Foundry.
+func IsPCF() bool {
+	// Minio PCF start-up script should set MINIO_PCF.
+	return os.Getenv("MINIO_PCF") != ""
```
Comment (harshavardhana) [inline cmd/update-main.go]: Is this is the only way to know? isn't there something PCF specific so that we don't have to add?

Comment (krishnasrinivas) [reply cmd/update-main.go]: These are the env vars set:
```
MONIT_DATE=Thu, 26 Oct 2017 00:16:53 +0000
MONIT_HOST=localhost
PATH=/bin:/usr/bin:/sbin:/usr/sbin
PWD=/etc/sv/monit
MONIT_PROCESS_PID=0
MONIT_EVENT=Started
MONIT_PROCESS_MEMORY=0
SHLVL=1
MONIT_PROCESS_CPU_PERCENT=0
MONIT_SERVICE=minio-server
MONIT_PROCESS_CHILDREN=0
MONIT_DESCRIPTION=Started
_=/usr/bin/printenv
```

But not sure if we should rely on the `MONIT_*` vars as I could not find documentation on these vars. Safer to depend on a var of our own.

Comment (harshavardhana) [reply cmd/update-main.go]: https://mmonit.com/monit/ is actually a different project, looks like there is no other way.

**Review by nitisht — DISMISSED** [review]
LGTM

File: cmd/update-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/update-main.go]: At least can we have a more MINIO_PIVOTAL_CLOUD_FOUNDRY - an expanded name to avoid any future confusion ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Minor comment env namspace can be protected and made more explanatory .. also avoiding any future conflicts.

**Review by harshavardhana — APPROVED** [review]

**Review by balamurugana — APPROVED** [review]
LGTM

**Pull request merged**